models/keyword_test.go

```diff
@@ -43,59 +45,75 @@ func TestKeywordDBTestSuite(t *testing.T) {
 }
 
 func (s *KeywordDBTestSuite) TestSaveKeywordsWithValidParams() {
-	keywordList := []Keyword{
-		{Keyword: "Hazard", UserID: s.userID},
-		{Keyword: "Ronaldo", UserID: s.userID},
-		{Keyword: "Neymar", UserID: s.userID},
-		{Keyword: "Messi", UserID: s.userID},
-		{Keyword: "Mbappe", UserID: s.userID},
+	nonAdwordLinks, _ := json.Marshal([]string{"test-non-ads-link"})
+	topPositionAdwordLinks, _ := json.Marshal([]string{"test-top-ads-link"})
+
+	keyword := Keyword{
+		Keyword:                 "Hazard",
+		Status:                  Pending,
+		LinksCount:              100,
+		NonAdwordsCount:         20,
+		NonAdwordLinks:          nonAdwordLinks,
+		TopPositionAdwordsCount: 5,
+		TopPositionAdwordLinks:  topPositionAdwordLinks,
+		TotalAdwordsCount:       25,
+		HtmlCode:                "test-html",
+		UserID:                  s.userID,
 	}
 
-	result, err := SaveKeywords(keywordList)
-
-	assert.Equal(s.T(), nil, err)
-	assert.Equal(s.T(), 5, len(result))
-	assert.Equal(s.T(), "Hazard", result[0].Keyword)
-	assert.Equal(s.T(), "Ronaldo", result[1].Keyword)
-	assert.Equal(s.T(), "Neymar", result[2].Keyword)
-	assert.Equal(s.T(), "Messi", result[3].Keyword)
-	assert.Equal(s.T(), "Mbappe", result[4].Keyword)
-}
+	result, err := SaveKeyword(keyword)
 
-func (s *KeywordDBTestSuite) TestSaveKeywordsWithEmptyStringSlice() {
-	keywordList := []Keyword{
-		{Keyword: "", UserID: s.userID},
-	}
+	var nonAdwordLinksVal []string
+	_ = json.Unmarshal(result.NonAdwordLinks, &nonAdwordLinksVal)
 
-	result, err := SaveKeywords(keywordList)
+	var topPositionAdwordLinksVal []string
+	_ = json.Unmarshal(result.TopPositionAdwordLinks, &topPositionAdwordLinksVal)
 
 	assert.Equal(s.T(), nil, err)
-	assert.Equal(s.T(), 1, len(result))
-	assert.Equal(s.T(), "", result[0].Keyword)
+	assert.Equal(s.T(), "Hazard", result.Keyword)
+	assert.Equal(s.T(), Pending, result.Status)
+	assert.Equal(s.T(), 100, result.LinksCount)
+	assert.Equal(s.T(), 20, result.NonAdwordsCount)
+	assert.Equal(s.T(), []string{"test-non-ads-link"}, nonAdwordLinksVal)
+	assert.Equal(s.T(), 5, result.TopPositionAdwordsCount)
+	assert.Equal(s.T(), []string{"test-top-ads-link"}, topPositionAdwordLinksVal)
+	assert.Equal(s.T(), 25, result.TotalAdwordsCount)
+	assert.Equal(s.T(), "test-html", result.HtmlCode)
 }
 
-func (s *KeywordDBTestSuite) TestSaveKeywordsWithEmptySlice() {
-	keywordList := []Keyword{}
+func (s *KeywordDBTestSuite) TestSaveKeywordsWithKeywordValue() {
```

Comment:
Shouldn't this be:

```suggestion
func (s *KeywordDBTestSuite) TestSaveKeywordsWithInvalidKeywordValue() {
```

Reply:
My bad. Fixed in 5c34fff8dc68b0f26cda6725b34aa03f4a711292